controller.xql

```diff
@@ -75,37 +76,47 @@ else if (ends-with($exist:resource, ".html") and starts-with($exist:path, "/pack
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
         <forward url="{concat($exist:controller, '/packages.html')}"/>
         <view>
-            <forward url="{concat($exist:controller, '/modules/view.xql')}">
+            <forward url="{concat($exist:controller, '/modules/view.xq')}">
                 <add-parameter name="abbrev" value="{substring-before($exist:resource, '.html')}"/>
             </forward>
         </view>
     </dispatch>
     
 else if (ends-with($exist:resource, ".html")) then
-    (: the html page is run through view.xql to expand templates :)
+    (: the html page is run through view.xq to expand templates :)
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
         <view>
-            <forward url="modules/view.xql">
+            <forward url="modules/view.xq">
                 <set-header name="Cache-Control" value="no-cache"/>
             </forward>
         </view>
     </dispatch>
 
+else if (contains($exist:path, "/public/") and ends-with($exist:resource, ".xar")) then
+    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
+        <forward url="../../public-repo-data/packages/{$exist:resource}"/>
+    </dispatch>
+
+else if (contains($exist:path, "/public/") and (ends-with($exist:resource, ".png") or ends-with($exist:resource, ".svg"))) then
+    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
+        <forward url="../../public-repo-data/icons/{$exist:resource}"/>
```

Comment:
Can we still have a variable collection where XARs are stored instead of a hard-coded path?

Reply:
@line-o I've now removed hardcoded collection names here and throughout the app. However, in this one instance in controller.xql, I haven't tried to resolve the arbitrary absolute path defined in config.xqm to the relative path needed in the forward directive.

Reply:
@line-o Also, I'd already changed scanrepo's file extension from xql to xqm.

Reply:
@line-o Ok, I've fixed this problem of the hard-coded relative path in controller.xql. See https://github.com/eXist-db/public-repo/pull/53/commits/b65a476d16148e5bc7bc6b13bf7d6b70124a5385.